packages/plugin/src/Bundles/Submissions/CipherFieldsBundle.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Solspace\Freeform\Bundles\Submissions;
+
+use Solspace\Commons\Helpers\PermissionHelper;
+use Solspace\Freeform\Elements\Submission;
+use Solspace\Freeform\Events\Submissions\CipherEvent;
+use Solspace\Freeform\Freeform;
+use Solspace\Freeform\Library\Bundles\FeatureBundle;
+use Solspace\Freeform\Library\Helpers\CipherHelper;
+use yii\base\Event;
+use yii\base\Exception;
+use yii\base\InvalidConfigException;
+
+class CipherFieldsBundle extends FeatureBundle
+{
+    protected CipherHelper $cipherHelper;
+
+    public function __construct(CipherHelper $cipherHelper)
+    {
+        $this->cipherHelper = $cipherHelper;
+
+        Event::on(
+            Submission::class,
+            Submission::EVENT_ENCRYPT_FIELDS,
+            [$this, 'encryptFields']
+        );
+
+        Event::on(
+            Submission::class,
+            Submission::EVENT_DECRYPT_FIELDS,
+            [$this, 'decryptFields']
+        );
+    }
```

Comment:
With PHP8.0 you can use constructor arguments to define object properties. To avoid having to write boilerplate.

```suggestion
class CipherFieldsBundle extends FeatureBundle
{
    public function __construct(private CipherHelper $cipherHelper)
    {
        Event::on(
            Submission::class,
            Submission::EVENT_ENCRYPT_FIELDS,
            [$this, 'encryptFields']
        );

        Event::on(
            Submission::class,
            Submission::EVENT_DECRYPT_FIELDS,
            [$this, 'decryptFields']
        );
    }
```